libc/test/src/sys/auxv/linux/getauxval_test.cpp

```diff
@@ -0,0 +1,27 @@
+//===-- Unittests for getaxuval -------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include "src/errno/libc_errno.h"
+#include "src/sys/auxv/getauxval.h"
+#include "test/UnitTest/ErrnoSetterMatcher.h"
+#include "test/UnitTest/Test.h"
+#include <src/string/strstr.h>
+#include <sys/auxv.h>
+
+using namespace LIBC_NAMESPACE::testing::ErrnoSetterMatcher;
+
+TEST(LlvmLibcGetauxvalTest, Basic) {
+  EXPECT_THAT(LIBC_NAMESPACE::getauxval(AT_PAGESZ), returns(GT(0ul)));
+  const char *filename;
+  auto getfilename = [&filename]() {
+    auto value = LIBC_NAMESPACE::getauxval(AT_EXECFN);
+    filename = reinterpret_cast<const char *>(value);
+    return value;
+  };
+  EXPECT_THAT(getfilename(), returns(NE(0ul)));
```

Comment:
is a lambda necessary here?

Reply:
I hope to set value to `filename` and then check with `returns`. Using a lambda does seem to be more clean to me.

Reply:
ah, right this is using the errno matcher as well